otx/hpo/utils.py

```diff
@@ -0,0 +1,124 @@
+        variable_name (Optional[str], optional): name of value. It's used for error message. Defaults to None.
+        error_message (Optional[str], optional): Error message to use when type is different. Defaults to None.
+
+    Raises:
+        TypeError: If type of value is different with the available type, then error is raised.
+    """
+    if not isinstance(value, available_type):  # type: ignore
+        if not isinstance(available_type, tuple):
+            available_type = [available_type]
+        if error_message is not None:
+            message = error_message
+        elif variable_name is not None:
+            message = (
+                f"{variable_name} should be "
+                + " or ".join([str(val) for val in available_type])
+                + f". Current {variable_name} type is {type(value)}"
+            )
+        else:
+            raise TypeError
+        raise TypeError(message)
+
+
+def check_positive(value: Any, variable_name: Optional[str] = None, error_message: Optional[str] = None):
+    """Validate that value is positivle.
+
+    Args:
+        value (Any): value to validate.
+        variable_name (Optional[str], optional): name of value. It's used for error message. Defaults to None.
+        error_message (Optional[str], optional): Error message to use when type is different. Defaults to None.
+
+    Raises:
+        ValueError: If value isn't positive, the error is raised.
+    """
+    if value <= 0:
+        if error_message is not None:
+            message = error_message
+        elif variable_name:
+            message = f"{variable_name} should be positive.\n" f"your value : {value}"
+        else:
+            raise ValueError
+        raise ValueError(message)
+
+
+def check_not_negative(value: Any, variable_name: Optional[str] = None, error_message: Optional[str] = None):
+    """Validate that value isn't negative.
+
+    Args:
+        value (Any): value to validate.
+        variable_name (Optional[str], optional): name of value. It's used for error message. Defaults to None.
+        error_message (Optional[str], optional): Error message to use when type is different. Defaults to None.
+
+    Raises:
+        ValueError: If value is negative, the error is raised.
+    """
+    if value < 0:
+        if error_message is not None:
+            message = error_message
+        elif variable_name:
+            message = f"{variable_name} should be positive.\n" f"your value : {value}"
+        else:
+            raise ValueError
+        raise ValueError(message)
+
+
+def check_mode_input(mode: str):
+    """Validate that mode is 'max' or 'min'.
+
+    Args:
+        mode (str): string to validate.
+
+    Raises:
+        ValueError: If 'mode' is not both 'max' and 'min', the error is raised.
+    """
+    if mode not in ["max", "min"]:
+        raise ValueError("mode should be max or min.\n" f"Your value : {mode}")
+
+
+def left_is_better(val1, val2, mode: str) -> bool:
```

Comment:
Suggest to change the name : better --> larger.

Reply:
Actually, it's for comparing score. the lower the better in some metric like loss and there is the other things like accuracy to the contrary. So how to let it be?

Reply:
Then, adding `metric` as a suffix is better option to avoid the confusion. `def left_is_better_metric`

Reply:
In my perspective, honestly, it's not confusing. Then how about to rename it to `left_value_is_better`?

Reply:
No objection